server/test/integration/vitalsTest/checkForInternalProblemsTest.js

```diff
@@ -0,0 +1,420 @@
+// Third-party dependencies
+const { expect, use } = require('chai')
+const { describe, it } = require('mocha')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+const rewire = require('rewire')
+
+// In-house dependencies
+const { ALERT_TYPE, factories, helpers } = require('brave-alert-lib')
+const db = require('../../../db/db')
+const { locationDBFactory, sessionDBFactory } = require('../../../testingHelpers')
+
+const vitals = rewire('../../../vitals')
+
+// Configure Chai
+use(sinonChai)
+
+const sandbox = sinon.createSandbox()
+
+const maxStillnessAlerts = parseInt(helpers.getEnvVar('MAX_STILLNESS_ALERTS'), 10)
+const intervalToCheckAlerts = parseInt(helpers.getEnvVar('INTERVAL_TO_CHECK_ALERTS'), 10)
+
+const currentDate = new Date()
+const changedTime = currentDate.getTime() - (intervalToCheckAlerts + 1) * 60 * 1000
+const oneMinuteBeforeIntervalDate = new Date(changedTime)
+
+describe('vitals.js integration tests: checkForInternalProblems', () => {
+  beforeEach(async () => {
+    await db.clearTables()
+    sandbox.stub(helpers, 'log')
+    sandbox.stub(helpers, 'logSentry')
+  })
+
+  afterEach(async () => {
+    await db.clearTables()
+    sandbox.restore()
+  })
+
+  describe('no locations in the database', () => {
+    beforeEach(async () => {
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('for a location that has no previous sessions', () => {
+    beforeEach(async () => {
+      // Insert a single location with no sessions
+      const client = await factories.clientDBFactory(db, { isSendingAlerts: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isSendingAlerts: true,
+      })
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('should not log to Sentry if exactly maxStillnessAlerts have occurred in set interval of time at a location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts sessions
+      const client = await factories.clientDBFactory(db, { isSendingAlerts: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isSendingAlerts: true,
+      })
+
+      this.incidentCategory = ALERT_TYPE.SENSOR_STILLNESS
```

Comment:
`ALERT_TYPE.SENSOR_STILLNESS` isn't an incidentCategory. However, you don't really care what the incidentCategory is, so you can just leave it out and the sessionDBFactory will fill in a valid value.

Reply:
oh that's very true, thank you for catching that! i think i got into the middle of changing it and then got distracted 😅